### PR TITLE
pythonPackages.pypandoc: 1.5.0 -> 1.6.3

### PR DESCRIPTION
**Pull request by sternenseemann**
* Changed upstream

* Test suite now works with latest pandoc

* pandoc --citeproc support

* Also substitute version in static-pandoc-path.patch

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pypandoc/default.nix [hunk 1]
```diff
@@ -4,22 +4,22 @@
...
     (substituteAll {
       src = ./static-pandoc-path.patch;
       pandoc = "${lib.getBin pandoc}/bin/pandoc";
+      pandocVersion = pandoc.version;
     })
     ./skip-tests.patch
-    ./new-pandoc-headings.patch
   ];
 
   checkInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pypandoc/default.nix:25]: Please update the homepage.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129653` at a4cff200 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>apostrophe</li>
    <li>python38Packages.pypandoc</li>
    <li>python39Packages.pypandoc</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/pypandoc/default.nix:17:5:
```
   |
17 |     (substituteAll {
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/pypandoc/default.nix:22:5:
```
   |
22 |     ./skip-tests.patch
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pypandoc/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/python-modules/pypandoc/default.nix:25]: done

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 129653 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>apostrophe</li>    <li>python38Packages.pypandoc</li>    <li>python39Packages.pypandoc</li>  </ul></details>

**Pull request merged**